camera-manager.js

```diff
@@ -121,53 +126,100 @@ const cameraManager = {
+
+      rayMatrix.lookAt(rayDirection,rayVectorZero,rayVectorUp);
+      rayQuaternion.setFromRotationMatrix(rayMatrix);
+
+      // Slightly move ray start position towards camera (to avoid hair,hat)
+      rayStartPos.copy(localPlayer.position);
+      rayStartPos.add( rayDirection.multiplyScalar(0.1) );
+
+      rayOriginArray[arrayIndex].copy(rayStartPos);
+      rayDirectionArray[arrayIndex].copy(rayQuaternion);
+
+    }
+
+    // Raycast from player postition with small offset
+    function initOffsetRayParams(arrayIndex,originPoint) {
+
+      rayDirection.subVectors(localPlayer.position,camera.position ).normalize();
+
+      rayMatrix.lookAt(rayDirection,rayVectorZero,rayVectorUp);
+      rayQuaternion.setFromRotationMatrix(rayMatrix);
+
+      rayOriginArray[arrayIndex].copy(originPoint);
+      rayDirectionArray[arrayIndex].copy(rayQuaternion);
+
+    }
 
-    let newVal = cameraOffsetTargetZ;
-    let hasIntersection = false;
 
     const localPlayer = metaversefile.useLocalPlayer();
 
-    let cornerPoints = [];
-    // Top left
-    cornerPoints[0] = new THREE.Vector3( -1, 1, ( camera.near + camera.far ) / ( camera.near - camera.far ) );
-    cornerPoints[0].unproject( camera );
+    
 
-    // Bottom left
-    cornerPoints[1] = new THREE.Vector3( -1, -1, ( camera.near + camera.far ) / ( camera.near - camera.far ) );
-    cornerPoints[1].unproject( camera );
+    let newVal = cameraOffsetTargetZ;
+    let hasIntersection = false;
 
-    // Top right
-    cornerPoints[2] = new THREE.Vector3( 1, 1, ( camera.near + camera.far ) / ( camera.near - camera.far ) );
-    cornerPoints[2].unproject( camera );
 
-    // Bottom right
-    cornerPoints[3] = new THREE.Vector3( 1, -1, ( camera.near + camera.far ) / ( camera.near - camera.far ) );
-    cornerPoints[3].unproject( camera );
+    // Camera - Top left 
+    initCameraRayParams(0,rayStartPos.set(-1, 1, ( camera.near + camera.far ) / ( camera.near - camera.far )).unproject( camera ));
 
+    // Camera - Bottom left
+    initCameraRayParams(1,rayStartPos.set(-1, -1, ( camera.near + camera.far ) / ( camera.near - camera.far )).unproject( camera ));
 
-    // Raycast from all camera corners
-    for(let i=0;i<4;i++) {
+    // Camera - Top right
+    initCameraRayParams(2,rayStartPos.set(1, 1, ( camera.near + camera.far ) / ( camera.near - camera.far )).unproject( camera ));
 
-      rayDirection.subVectors(localPlayer.position, cornerPoints[i] ).normalize();
+    // Camera - Bottom right
+    initCameraRayParams(3,rayStartPos.set(1, -1, ( camera.near + camera.far ) / ( camera.near - camera.far )).unproject( camera ));
 
-      rayMatrix.lookAt(rayDirection,rayVectorZero,rayVectorUp);
-      rayQuaternion.setFromRotationMatrix(rayMatrix);
+    // Player postition - offset to left
+    rayStartPos.copy(localPlayer.position);
+    rayOffsetPoint.set(-1, 0, 0);
+    rayOffsetPoint.applyQuaternion(camera.quaternion);
```

Comment:
I would have chained these variables for less typing.

Almost everything in three.js returns itself,

```
rayOffsetPoint.set(-1, 0, 0)
  .applyQuaternion(camera.quaternion);
```